IFC models: Add routes for split and viewer

oliverguenther opened the pull request:
- Add routes for viewer only, list only, viewer and list
- Uses named uiviews to dynamically mount a component in a tree.

oliverguenther committed:
Add routes for split and viewer

ulferts committed:
recreate names for routes

ulferts committed:
rename page into container

HDinger committed:
Open isolated query space in base component so that is available in the IfcViewerPageComponent

openprojectci commented:
<!--
  0 Errors
  3 Warnings: <a href="https://github.com/op...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          3 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/opf/openproject/blob/2a9790c93329f82deb4157bcf35b11f8a0b70184/frontend/src/app/modules/calendar/wp-calendar-entry/wp-calendar-entry.component.ts#L32">frontend/src/app/modules/calendar/wp-calendar-entry/wp-calendar-entry.component.ts#L32</a> - Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/opf/openproject/blob/2a9790c93329f82deb4157bcf35b11f8a0b70184/frontend/src/app/modules/grids/grid/grid.component.ts#L37">frontend/src/app/modules/grids/grid/grid.component.ts#L37</a> - Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/opf/openproject/blob/2a9790c93329f82deb4157bcf35b11f8a0b70184/frontend/src/app/modules/ifc_models/ifc-viewer/ifc-viewer.component.ts#L35">frontend/src/app/modules/ifc_models/ifc-viewer/ifc-viewer.component.ts#L35</a> - Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

ulferts committed:
query space for ifc page

We add the WorkPackagesBaseComponent as a ui router parent to gain a page wide query space. The IFC Page then inherits from the abstract WorkPackagesViewBase component to e.g get acces to the query params handling.

ulferts committed:
Revert "Open isolated query space in base component so that is available in the IfcViewerPageComponent"

This reverts commit dd24278b4805b1a39b331c9fa1e0d2e29fb5315b.

ulferts committed:
fix height of the embedded list

ulferts committed:
remove trailing slash

ulferts committed:
remove unfinished view toggle component

ulferts commented:
I believe the PR to be mergeable @HDinger, @oliverguenther. There are three valid TODOs that we need to take a look at but that can happen later on. I also did not add tests for switching between the various states, so the states are not yet completely covered. 

But we should be able to work from there.

HDinger reviewed (approved):
Looks good 👍

ulferts committed:
Merge branch 'dev' into bim/feature/ifc-routes
